packages/editor/src/functions/EditorControlFunctions.ts

```diff
@@ -569,8 +569,11 @@ const reparentObject = (entities: Entity[], before?: Entity | null, parent = get
             (n) => n.extensions?.[UUIDComponent.jsonID] === getComponent(before, UUIDComponent)
           )
           gltf.data.scenes![0].nodes.splice(beforeIndex, 0, nodeIndex)
+          gltf.data.nodes?.splice(beforeIndex, 0, gltf.data.nodes?.[nodeIndex])
```

Comment:
what are these extra lines doing exactly?

Reply:
When the node is reparented in scene root, the order previously changed only in `gltf.data.scenes`. However, the hierarchy panel is dependent on change in `gltf.data.nodes` for [hierarchy change detection](https://github.com/EtherealEngine/etherealengine/blob/0e1e55d8809128472211a42e5d4efc4a66df7324/packages/ui/src/components/editor/panels/Hierarchy/container/index.tsx/#L204-L207) (tried to do it in the hierarchy panel itself for detecting this change but it becomes too complex and requiring almost a full refactor of our change detection and tree walking algorithm). 
Since we are [changing the order in nodes' children when it is not the parent](https://github.com/EtherealEngine/etherealengine/blob/0e1e55d8809128472211a42e5d4efc4a66df7324/packages/editor/src/functions/EditorControlFunctions.ts#L585>), we should also change it in the gtlf nodes' array

Reply:
Gotcha. We definitely want to do what makes the most sense for the GLTF data, it is the fundamental data that should describe how the hierarchy panel displays objects. Ie, the hierarchy panel should show exactly what is in the GLTF, so this makes sense.